packages/engine/lib/error/src/frame.rs

```diff
@@ -33,10 +40,27 @@ use crate::{
 pub struct Frame {
     inner: ManuallyDrop<Box<FrameRepr>>,
```

Comment:
I moved the logic to one function rather than having it at four functions. The SAFETY section were expanded: d7e784641c7b24d0da494aeab0ad37f2f686532d

Reply:
I think I'd still kind of like to see a comment here saying why it's `ManuallyDrop` but if you're happy with it being on the actual `unsafe` portion then you can resolve this comment.

Reply:
The SAFETY section on an unsafe block should not explain why this is unsafe but why is it safe to call the function inside the unsafe block. The `FrameRepr` must not be dropped without using the vtable, and this is exactly why we use ManuallyDrop: We have a custom drop implementation that takes care of this.

Reply:
Which is why I'm not suggesting we necessarily add a "SAFETY" comment on the `inner` but it might be worth saying why we need `ManuallyDrop` at all on the `FrameRepr`. Was just saying we could add a comment exactly like you just wrote:
> The FrameRepr must not be dropped without using the vtable, so it's wrapped in ManuallyDrop, there's a custom drop implementation that takes care of this

Anyway the code somewhat speaks for itself, if someone wants to find the reasoning I'm sure they can infer it from finding `impl Drop`, I'm just saying we could save them the look

Reply:
Ah, now I get you 😄  Changed it again